server/src/main/java/org/jboss/as/server/moduleservice/ModuleDefinition.java

```diff
@@ -25,16 +25,31 @@ public class ModuleDefinition {
     private final Set<ModuleDependency> dependencies;
     private final ModuleSpec moduleSpec;
 
+
+    /** @deprecated use {@link #ModuleDefinition(String, Set, ModuleSpec)} */
+    @Deprecated(forRemoval = true)
     public ModuleDefinition(final ModuleIdentifier moduleIdentifier, final Set<ModuleDependency> dependencies, final ModuleSpec moduleSpec) {
         this.moduleIdentifier = moduleIdentifier;
         this.dependencies = dependencies;
         this.moduleSpec = moduleSpec;
     }
 
+    public ModuleDefinition(final String moduleIdentifier, final Set<ModuleDependency> dependencies, final ModuleSpec moduleSpec) {
+        this.moduleIdentifier = ModuleIdentifier.fromString(moduleIdentifier);  // inefficient but this is unused. When we switch the use to this we'll store the string.
+        this.dependencies = dependencies;
+        this.moduleSpec = moduleSpec;
+    }
+
+    /** @deprecated use {@link #getModuleName()}  */
+    @Deprecated(forRemoval = true)
     public ModuleIdentifier getModuleIdentifier() {
         return moduleIdentifier;
     }
 
+    public String getModuleName() {
+        return moduleIdentifier.toString(); // inefficient but this is unused. When we switch the use to this we'll store the string.
+    }
```

Comment:
I wonder if getModuleName is what we would want. The ModuleIdentifier is composed by `name` and `slot` attributes. So `getModuleName()` returning the result of `moduleIdentifier.toString()` maybe is not what we want since it could drive into confusion since someone could be expecting to get just the ModuleIdentifier name. 

As additional detail, the JBoss Modules documentation still makes a distinction about what a module name is (https://jboss-modules.github.io/jboss-modules/manual/#names) and what are module identifiers https://jboss-modules.github.io/jboss-modules/manual/#slots

Reply:
getModuleIdentifier() is already taken, so ....